### ISSUE TITLE
Loader.Error not changing on Exception

### ISSUE THREAD

**Issue by clifzero** [roubachof/Sharpnado.TaskLoaderView#8]
Hi Jean

I don't think this is a bug in your library, might just be a misunderstanding on my part. Earlier today I was simulating a scenario whereby a user doesn't have an internet connection, wifi's off, mobile data on but they don't have data. I set the HttpClient's timeout to 30 seconds, when the TaskCanceled exception is thrown, there's no property changed event from the TaskLoaderNotifier like how there usually is for other errors, I usually `check Loader.Error` and  handle errors based on what's there.  Am I missing anything?

Here's a [sample](https://www.dropbox.com/s/kazmc8b047bmvuq/sample_application.zip?dl=0) that replicates the issue

Thanks again for the library Jean!

**Comment by roubachof** [comment]
I think cancel is a task state. Hopefully you can configure the TaskMonitor (which is used by the task loader notifier) to consider the cancel task as error. Have a look at the readme of the TaskMonitor.

**Comment by clifzero** [comment]
100% right it is a state, I'll have a look and let you know if I can setup a global configuration for task loader through task monitor. Thanks Jean!

**Comment by clifzero** [comment]
Thank you Jean, I had a look and there is a property to get the job done (TaskMonitorConfiguration.ConsiderCanceledAsFaulted). With this set to true, I get a property changed event for IsFaulted.

Thank you for your quick replies by the way!

You're a Legend!

**Comment by clifzero** [comment]
I think I might have closed the issue a bit too soon, so here's what's happening now, 

Once I see that the task is faulted because of a cancellation due to me setting TaskMonitorConfiguration.ConsiderCanceledAsFaulted to true, I then manually set the Loader.Error (because the label in my error view binds to Loader.Error), in the OnTaskFaulted method , Loader.Error is then being set to faultTask.InnerException, which is null🤷🏻‍♂️, any suggestions?

**Comment by roubachof** [comment]
I would bind the error view directly to the notifier and have a NotifierToErrorMessageConverter. If IsCanceled is true you can set an error message accordingly. Otherwise you determine your error message from your Error field. But error should contain the cancelled exception, so I have to fix this.

**Comment by clifzero** [comment]
Everything you mentioned above is exactly what I'm doing. In the meantime let me show you my ugly workaround 😁

```
if (e.PropertyName == "Error")
            {
                if (Loader.Error != null)
                {
                    if (Loader.IsFaulted && ! Loader.IsCanceled)
                    {
                        TaskMonitor.Create(HandleErrorAsync);
                     }
                }

                else if (Loader.Error == null)
                {
                    if (Loader.IsFaulted && Loader.IsCanceled)
                    {
                        Loader.Error = new TaskCanceledException();
                    }
                }


            }
            if (e.PropertyName == "IsFaulted")
            {
                if (Loader.IsFaulted && Loader.IsCanceled)
                {
                    Loader.Error = new Exception();
                }
            }
```

**Comment by clifzero** [comment]
Hi Jean

I hope you're doing well, any updates on this as yet? Did you manage to find time to resolve the issue?

**Comment by roubachof** [comment]
Hey man !
Was fixed in code, I will release a bunch of updated packages soon :)

**Comment by clifzero** [comment]
Thanks man! That’ll be highly appreciated there is another issue I found though, but I’m afraid I’ll have to open another issue for that bug

**Comment by roubachof** [comment]
Ok please open it quickly cause I'm working on it right now and will be quick and easy to fix. If it's later, dunno when it will be fixed.

**Comment by Cliffy786** [comment]
Ok will do, just give me a small moment please.From: Jean-Marie AlfonsiSent: Friday, 18 September 2020 17:36To: roubachof/Sharpnado.TaskLoaderViewCc: Clifton Steenkamp; State changeSubject: Re: [roubachof/Sharpnado.TaskLoaderView] Loader.Error not changing on Exception (#8) Ok please open it quickly cause I'm working on it right now and will be quick and easy to fix. If it's later, dunno when it will be fixed.—You are receiving this because you modified the open/close state.Reply to this email directly, view it on GitHub, or unsubscribe.

**Comment by clifzero** [comment]
I've logged the new issue, Let me know how it goes.

**Comment by roubachof** [comment]
should be fixed with the v2.2.0 attached in issue #9
